__tests__/courseController.tests.mjs

```diff
@@ -0,0 +1,122 @@
+import admin from '../server/config/firebase.mjs';
+import {
+  createCourse,
+  getCourse,
+  updateCourse,
+  deleteCourse
+} from '../server/controllers/courseController.mjs';
+import { USERS, COURSES } from '../server/controllers/constants.mjs';
+import supertest from 'supertest';
+import express from 'express';
+
+const app = express();
+app.use(express.json());
+
+// Middleware to simulate authenticated user
+const simulateAuth = (req, res, next) => {
+  req.user = { uid: 'testUserId' }; 
+  next();
+};
+
+// Route setup for course endpoints
+app.post('/course', simulateAuth, createCourse);
+app.get('/course/:uid', simulateAuth, getCourse);
+app.patch('/course/:uid', simulateAuth, updateCourse);
+app.delete('/course/:uid', simulateAuth, deleteCourse);
+
```

Comment:
There are additional endpoints in the courseController.mjs file, such as getAllCourses, getTeacherCourses, and getUserCourses. You might need to add more test cases for these.

Reply:
These new endpoints weren't in `courseController.mjs` when I was working on my tests. I'll leave this file as is for now because I see you have some upcoming changes for the courseController.mjs in your new PR. Once those changes are merged, I'll update the test files accordingly.